test/test_config_parser.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require 'test/unit'
+require 'config_parser'
+require 'config_pb'
+require 'fluent/config/v1_parser'
+
+# tests for config tool
+class TestConfigParser < Test::Unit::TestCase
+  def test_comments_not_parsed
+    expected = Config::Directive.new(
+      name: 'ROOT',
+      directives: [
+        Config::Directive.new(name: 'system', params: [get_param('rpc', '0.0.0:2')]),
+        Config::Directive.new(name: 'source', params: [get_param('@type', 'forward')])
+      ]
+    )
+    assert(ConfigParser.proto_config(ConfigParser.parse_config('test/data/comments.conf')) == expected)
+  end
+
+  def test_special_characters_parsed_correctly
+    expected = Config::Directive.new(
+      name: 'ROOT',
+      directives: [Config::Directive.new(name: 'source', params: create_params_special)]
```

Comment:
Also, could you inline create_params_special (and the other similar functions)?  I think it's easier if we can read the whole output in one place.  Something like this:

    expected = Config::Directive.new(
      name: 'ROOT',
      directives: [
        Config::Directive.new(
          name: 'source',
          params: [
            get_param('@type', 'tail'),
            get_param('source_host_key', 'host'),
            get_param('tag', 'test'),
            get_param('bind', '0.0.0.0'),
            get_param('dummy', '[{"message":"hello"},{"message":"bye"}]'),
            get_param('dummy2', '{"message":"again"}')
          ]
        )
      ]
    )

What do you think?

Reply:
I would ideally like to do that too, but when I use rubocop it won't let me define functions that exceed 10 lines of code. Is there any way to change that? Because if I'm limited to 10 lines, I won't be able to make everything inline.